src/common/templates/_ingress.tpl

```diff
@@ -26,6 +26,9 @@ metadata:
     {{- if $.Values.global.ingress.objects.annotations }}
     {{- include "harnesscommon.tplvalues.render" (dict "value" $.Values.global.ingress.objects.annotations "context" $) | nindent 4 }}
     {{- end }}
+    {{- if $.Values.ingress.serviceUpStream.annotations }}
```

Comment:
Not recommended like this:
add a field under ingress objects which are under conditions. like conditionalAnnotations
which expects an array of annotations and conditions associated with it

```
ingress:
  objects:
    - annotations: 
        nginx.ingress.kubernetes.io/rewrite-target: /$2
       conditionalAnnotations: <-- example
       - newAnnotation: newValue
          condition: "global.upstream.enabled"
      paths:
      - path: '{{ .Values.global.ingress.pathPrefix }}/pipeline(/|$)(.*)'

 ```

Reply:
made changes as per your request. please validate.